### PR TITLE
fix: update to support hapi 17

### PR DESCRIPTION
**Pull request by Saeris**
Hapi 17 introduces breaking changes to it's API which affects all existing plugins. This fix is intended to migrate from hapi 16 to 17. Currently `graphql-playground-middleware-hapi` will throw an error such as the following:

```sh
error: AssertionError [ERR_ASSERTION]: Invalid plugin options {
  "plugin": {
    "register": function (server, options, next) {\n    if (arguments.length !== 3) {\n        throw new Error(\"Playground middleware expects exactly 3 arguments, got \" + arguments.length);\n    }\n    var path = options.path, _a = options.route, config = _a === void 0
 ? {} : _a, rest = __rest(options, [\"path\", \"route\"]);\n    var middlewareOptions = __assign({}, rest, { version: pkg.version });\n    server.route({\n        method: 'GET',\n        path: path,\n        config: config,\n        handler: function (request, reply) {\n
            reply(graphql_playground_html_1.renderPlaygroundPage(middlewareOptions)).header('Content-Type', 'text/html');\n        },\n    });\n    return next();\n},
    "options": {
      "path": "/playground",
      "endpoint": "/graphql"
    },
    "name" [1]: -- missing --
  }
}

[1] "name" is required
```

Changes:

Updated `graphql-playground-middleware-hapi` plugin definition to support hapi 17.

Removed `attributes?: any` and replaced it with `pkg?: any`. `multiple` is `false` by default in hapi 17.

Please see the following for more information on the changes between hapi 16 and 17:

https://futurestud.io/tutorials/hapi-v17-upgrade-guide-your-move-to-async-await
https://hapijs.com/api#plugins

Updated the example project to use hapi 17, including dependencies.

Please change package version numbers as required.

### CODE REVIEW

File: packages/graphql-playground-middleware-hapi/examples/basic/index.js [hunk 1]
```diff
@@ -40,19 +39,30 @@ server.register({
...
+async function start() {
+  console.log(`Setting up server...`)
+  try {
+    await server.register(plugins)
+    await server.start()
+    console.log(`Server running at: ${server.info.uri}`)
+  } catch (err) {
+    console.log(`Failed to start server!`, err)
   }
-  console.log(`Server running at: ${server.info.uri}`)
```
Comment (Saeris) [inline packages/graphql-playground-middleware-hapi/examples/basic/index.js]: Whoops! This line can be deleted!

**Comment by schickling** [comment]
@timsuchanek can you merge & publish this?

**Comment by Saeris** [comment]
I'll be heading out of town later today and won't be available again until about the middle of next week. As suggested I'll check out `npm publish` when I get back to test this and update you then.

**Comment by Saeris** [comment]
quick update:

I found a bit of time to try publishing it to my scope to test as you suggested, found I needed to make some more changes.

Running into an issue with the middleware now where it can't find the css/middleware.js on the CDN and I think it's because of a version number mismatch. I'll look into a fix later on.

Seems likely that because the mono-repo version numbers need to be in-sync to dynamically grab the version number from the package.json file, that's one part of the problem since my version of the package will always been out of sync with the mono-repo.

I can test to make sure the middleware works I think by using a specific version number that's on the CDN. For now I can at least get the page to show a loading indicator. Here's the console errors from Chrome for reference:

```sh
GET http://cdn.jsdelivr.net/npm/graphql-playground-react@1.3.6/build/static/css/index.css net::ERR_ABORTED
GET http://cdn.jsdelivr.net/npm/graphql-playground-react@1.3.6/build/static/js/middleware.js net::ERR_ABORTED
Uncaught ReferenceError: GraphQLPlayground is not defined
    at playground:531
```

I'll push my WIP in case anyone has the time to take a look.

File: packages/graphql-playground-middleware-hapi/src/index.ts [hunk 2]
```diff
@@ -9,45 +9,40 @@ import {
 const pkg = require('../package.json')
 
 export interface Register {
-  (server: Server, options: MiddlewareOptions, next: any): void
-  attributes?: any
+  (server: Server, options: MiddlewareOptions): void
```
Comment (Saeris) [inline packages/graphql-playground-middleware-hapi/src/index.ts]: Not too familiar with Typescript, so if I'm doing this wrong please let me know!

File: packages/graphql-playground-middleware-hapi/examples/basic/index.js [hunk 3]
```diff
@@ -1,13 +1,17 @@
 const hapi = require('hapi')
 const { graphqlHapi } = require('apollo-server-hapi')
-const hapiPlayground = require('graphql-playground-middleware-hapi').default
+const hapiPlayground = require('@saeris/graphql-playground-middleware-hapi').default
```
Comment (timsuchanek) [inline packages/graphql-playground-middleware-hapi/examples/basic/index.js]: Please change this to `../../dist/index`

File: packages/graphql-playground-middleware-hapi/src/index.ts [hunk 4]
```diff
@@ -9,45 +9,40 @@ import {
...
+    const middlewareOptions: RenderPageOptions = {
+      ...rest,
+      version: '1.3.6' || pkg.version,
+    }
+
+    server.route({
+      method: 'GET',
+      path,
+      config,
+      handler: async (request, h) => h.response(await renderPlaygroundPage(middlewareOptions)).type('text/html')
```
Comment (timsuchanek) [inline packages/graphql-playground-middleware-hapi/src/index.ts]: With the latest `graphql-playground-html` there is no need for async await anymore

**Pull request merged**